src/processor/reconnection-graph.service.ts

```diff
@@ -430,12 +419,75 @@ export class ReconnectionGraphService {
+    dsnpUserId: MessageSourceId,
+    provideId: ProviderId,
+    providerKeys: KeyringPair,
+    graphKeyPairs: ProviderKeyPair[],
+    batch: SubmittableExtrinsic<'rxjs', ISubmittableResult>[]
+    ): Promise<void> {
+    try {
+      const [event, eventMap] = await this.blockchainService.createExtrinsic(
+        { pallet: 'frequencyTxPayment', extrinsic: 'payWithCapacityBatchAll' }, 
+        { eventPallet: 'utility', event: 'BatchCompleted' },
+        providerKeys, batch).signAndSend();
+      
       if (!event) {
-        throw new Error(`Error submitting extrinsic`);
+        // if we dont get any events, covering any unexpected connection errors
+        throw new Error(`No events were found for ${dsnpUserId.toString()}`);
+      }
+
+      if(!this.blockchainService.api.events.utility.BatchCompleted.is(event)) {
+        this.logger.warn(`Batch failed event found for ${dsnpUserId.toString()}: ${event}`);
+        if(this.blockchainService.api.events.utility.BatchInterrupted.is(event)) {
+          // Since frequency extrinsics does pre-flight checks via signed extension, we should not see any other events
+          // Any error from such pre-flight checks are caught in the catch block
+          // BatchInterrupted is caused during tx execution and any error arising from
+          // stateful storage would mean our configuration is wrong
+          this.logger.warn(`Unexpected event found for ${dsnpUserId.toString()}`);
+          this.logger.warn(event);
+          this.logger.warn(eventMap);
+          throw new Error(`Batch interrupted event found for ${dsnpUserId.toString()}`);
+        }
+      }
+    } catch (e) {
+      this.logger.error(`Error processing batch for ${dsnpUserId.toString()}: ${e}`);
+      //Following errors includes are checked against
+      // 1. Inability to pay some fees`
+      // 2. Transaction is not valid due to `Target page hash does not match current page hash`
+
+      if (e instanceof Error && e.message.includes('Inability to pay some fees')) {
+        // in case capacity is low pause the queue
+        this.graphUpdateQueue.pause();
+        throw e;
+      } else if (e instanceof Error && e.message.includes('Target page hash does not match current page hash')) {
+        // in case page hash does not match, re-import the user's DSNP Graph from the blockchain and form import bundles
+        // import bundles are used to import the user's DSNP Graph into the graph SDK
+        // here we update the state of the graph with the latest data from the chain
+        // once state is updated we do a non-transitive graph update
+        await this.importBundles(dsnpUserId, graphKeyPairs);
+        // refresh state and queue a non-transitive graph update
+        // this is safe to do as we are only updating single user's graph
+        const { key: jobId, data } = createGraphUpdateJob(dsnpUserId, provideId, SkipTransitiveGraphs);
+        this.graphUpdateQueue.add('graphUpdate', data, { jobId });
```

Comment:
The signed extension throws error before tx makes it to pool which is caugh here to handle two cases
* Fee too low: we should do a time pause? for now it will just pause the queue
* Page state is stale, in that case since we sending batches for only one user at a time, we can imply re-import and queue a non-transitive job so it gets cleared up in next run